xds/src/main/java/io/grpc/xds/XdsLrsClient.java

```diff
@@ -0,0 +1,325 @@
+      this.stub = checkNotNull(stub, "stub");
+      this.stopwatch = checkNotNull(stopwatch, "stopwatch");
+    }
+
+    void start() {
+      lrsRequestWriter = stub.withWaitForReady().streamLoadStats(this);
+    }
+
+    @Override
+    public void onNext(final LoadStatsResponse response) {
+      syncContext.execute(new Runnable() {
+        @Override
+        public void run() {
+          handleResponse(response);
+        }
+      });
+    }
+
+    @Override
+    public void onError(final Throwable t) {
+      syncContext.execute(new Runnable() {
+        @Override
+        public void run() {
+          handleStreamClosed(Status.fromThrowable(t)
+              .augmentDescription("Stream to XDS management server had an error"));
+        }
+      });
+    }
+
+    @Override
+    public void onCompleted() {
+      syncContext.execute(new Runnable() {
+        @Override
+        public void run() {
+          handleStreamClosed(
+              Status.UNAVAILABLE.withDescription("Stream to XDS management server was closed"));
+        }
+      });
+    }
+
+    private void sendLoadReport() {
+      long interval = stopwatch.elapsed(TimeUnit.NANOSECONDS);
+      stopwatch.reset().start();
+      // TODO: get load report and send to management server
+      lrsRequestWriter.onNext(LoadStatsRequest.newBuilder()
+          .setNode(Node.newBuilder()
+              .setMetadata(Struct.newBuilder()
+                  .putFields(
+                      TRAFFICDIRECTOR_GRPC_HOSTNAME,
+                      Value.newBuilder().setStringValue(serviceName).build())))
+          .addClusterStats(ClusterStats.newBuilder().setLoadReportInterval(
+              numToDuration(interval, TimeUnit.NANOSECONDS)))
+          .build());
+      scheduleNextLoadReport();
+    }
+
+    private void scheduleNextLoadReport() {
+      if (loadReportIntervalNano > 0) {
+        // Cancel pending load report and reschedule with updated load reporting interval.
+        if (loadReportTimer != null && loadReportTimer.isPending()) {
+          loadReportTimer.cancel();
+        }
+        loadReportTimer = syncContext.schedule(
+            new LoadReportingTask(this), loadReportIntervalNano, TimeUnit.NANOSECONDS,
+            timerService);
+      }
+    }
+
+    private void handleResponse(LoadStatsResponse response) {
+      if (closed) {
+        return;
+      }
+      logger.log(ChannelLogLevel.DEBUG, "Got an LRS response: {0}", response);
+
+      if (!initialResponseReceived) {
+        initialResponseReceived = true;
+      }
+      loadReportIntervalNano
+          = durationToNum(response.getLoadReportingInterval(), TimeUnit.NANOSECONDS);
+      List<String> serviceList = Collections.unmodifiableList(response.getClustersList());
+      // For gRPC use case, LRS response will only contain one cluster, which is the same as in
+      // the EDS response.
+      checkState(serviceList.size() == 1, "Excessive reporting cluster: %s", serviceList);
```

Comment:
This will thrown to the caller, which is synchronization context, and crash the channel. The general rule is, don't let external input error crash your program. Such error should be handled gracefully, for example, you can log an ERROR to the logger, then skip the call to `scheduleNextLoadReport()`.

Reply:
ACK.